src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,8 +395,247 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
+  std::unordered_map<std::string, const Vector<BaseFloat> * > ivectors_;
+
+  // online_ivector_feats_ is the iVectors if we're using online-estimated ones.
+  std::unordered_map<std::string,
+    const Matrix<BaseFloat> * > online_ivector_feats_;
+
+  // online_ivector_period_ helps us interpret online_ivector_feats_; it's the
+  // number of frames the rows of ivector_feats are separated by.
+  int32 online_ivector_period_;
+
+  // an object of CachingOptimizingCompiler. For speed, except for "shorter
+  // than chunk size" batch, we always get the ComputationRequest from RqType_.
+  // Then we use CachingOptimizingCompiler to compiler the CompuationRequest
+  // once, when it was first used.
+  CachingOptimizingCompiler compiler_;
+
+  // The current log-posteriors that we got from the last time we
+  // ran the computation.
+  std::unordered_map<std::string, Matrix<BaseFloat>* > log_post_;
+
+  // note: num_subsampled_frames_ will equal feats_.NumRows() in the normal case
+  // when opts_.frame_subsampling_factor == 1.
+  std::unordered_map<std::string, int32> num_subsampled_frames_;
+
+  std::unordered_map<std::string, bool> is_computed_;
+  // store each utterance id in order. We don't use a queue for here as,
+  // in function "compute()" which is a blocking call, we will keep on
+  // computing without taking the results out.
+  std::list<std::string> utt_list_;
+
+  // The meaning of the tuple is
+  // (utt_id, first_input_frame_index, last_input_frame_index,
+  // first_output_subsampled_index, last_output_frame_index,
+  // output_offset)
+  // The first/last_input_frame_index is used to fill up the input feature.
+  // The first/last_output_subsampled_frame_index is used to fill up the output
+  // posterior. The output_offset is useful in overleap with preivous chunk
+  // case. It is used to transit the output index.
+  typedef std::tuple<std::string, int32, int32, int32, int32, int32> BatchInfo;
+  typedef std::list<BatchInfo> BatchInfoQueue;
+
+  // store the information of the current batch. The key is pair
+  // (tot_left_context, tot_right_context) which would equal the model
+  // left/right context plus the extra left/right context. Each key corrsponds
+  // to a kind of batch. When ensure_exact_final_context is true, (-1, -1) will
+  // indexes those "shorter than chunk size" utterances.
+  typedef std::unordered_map<std::pair<int32, int32>, BatchInfoQueue*,
+                             IntPairHasher, IntPairEqual> BatchInfoMap;
+  BatchInfoMap batch_info_;
+  BatchInfo last_batch_info_;
+
+  // The key is (tot_left_context, tot_right_context), which would equal the
+  // model left/right context plus the extra left/right context. The value is
+  // corresponding ComputationRequest pointer. It is updated in function
+  // PrepareCompuationRequest().
+  typedef std::unordered_map<std::pair<int32, int32>, ComputationRequest *,
+                             IntPairHasher, IntPairEqual> NormalRqType;
+  NormalRqType RqType_;
```

Comment:
This variable name RqType_ is not style-guide compliant.  Maybe you could call it something like `context_to_request_`.  And I don't like the typename NormalRqType.  Perhaps ComputatationRequestMap would be better.